sympy/matrices/sparse.py

```diff
@@ -491,6 +491,10 @@ def _mat(self):
         in DenseMatrix use `_mat` directly to speed up operations."""
         return list(self)
 
+    @property
+    def _flat(self):
+        return list(self)
```

Comment:
Is this better than making it an attribute that is set to `tuple(self)`? If I understand correctly, every call to `_flat` will return a unique, mutable copy of self's elements.

Reply:
Yes, returning a tuple might be better.

What isn't clear to me is how much we need to respect backwards compatibility wrt `_mat` which is technically private but is also used widely in the SymPy codebase and potentially other downstream code as well. It isn't possible to maintain full backwards compatibility because `_mat` can be modified directly as a way of mutating the Matrix. If we take out the possibility that someone is using `_mat` to mutate the Matrix then returning a list is best for backwards compatibility.

But what if someone is mutating the list? Should we raise a warning or make sure that the operation completely fails (which would be the case if a tuple was returned)?

Reply:
Actually for mutable matrices we need to construct the list/tuple each time. In the case of immutable matrices a tuple could be stored. Perhaps it could be a tuple in all cases even though it would have to be created fresh each time for mutable matrices.

The performance here should not matter much in the end since the intention is that DomainMatrix would end up handling each of the operations that currently accesses `._mat`.